python/ml4ir/base/features/feature_config.py

```diff
@@ -59,25 +19,81 @@ class FeatureConfigKey:
 
 class FeatureConfig:
     """
-    Class to store features to be used for the Relevance models
+    Class that defines the features and their configurations used for
+    training, evaluating and serving a RelevanceModel on ml4ir.
+
+    Attributes
+    ----------
+    features_dict : dict
+        Dictionary of features containing the configuration for every feature
+        in the model. This dictionary is used to define the FeatureConfig
+        object.
+    logger : `Logging` object
+        Logging handler to log progress messages
+    query_key : dict
+        Dictionary containing the feature configuration for the unique data point
+        ID, query key
+    label : dict
+        Dictionary containing the feature configuration for the label field
+        for training and evaluating the model
+    features : list of dict
+        List of dictionaries containing configurations for all the features
+        excluding query_key and label
+    all_features : list of dict
+        List of dictionaries containing configurations for all the features
+        including query_key and label
+    train_features : list of dict
+        List of dictionaries containing configurations for all the features
+        which are used for training, identified by `trainable=False`
+    metadata_features : list of dict
+        List of dictionaries containing configurations for all the features which
+        are NOT used for training, identified by `trainable=False`. These can be
+        used for computing custom losses and metrics.
+    features_to_log : list of dict
+        List of dictionaries containing configurations for all the features which
+        will be logged when running `model.predict()`, identified using
+        `log_at_inference=True`
+    group_metrics_keys : list of dict
+        List of dictionaries containing configurations for all the features
+        which will be used to compute groupwise metrics
+    secondary_labels : list of dict
+        List of dictionaries containing configurations for all the features
+        which will be used as secondary labels to compute secondary metrics.
+        The implementation of the secondary metrics and the usage of the secondary
+        labels is up to the users of ml4ir
+
+    Notes
+    -----
+    Abstract class that is overriden by ExampleFeatureConfig and
+    SequenceExampleFeatureConfig for the respective TFRecord types
     """
 
     def __init__(self, features_dict, logger: Optional[Logger] = None):
+        """
```

Comment:
it is hard to find the doc of the ctor
the doc of the ctor should be "higher" in the html output

same remark for other classes defined in this file

![image](https://user-images.githubusercontent.com/63343516/95366028-2b268b00-08d3-11eb-89ba-af05c29040ca.png)